ppdet/data/transform/operator.py

```diff
@@ -0,0 +1,1738 @@
+                    sample['gt_keypoint'], width)
+            sample['flipped'] = True
+            sample['image'] = im
+        return sample
+
+
+@register_op
+class ResizeOp(BaseOperator):
+    def __init__(self, target_size, keep_ratio=True, interp=cv2.INTER_LINEAR):
+        """
+        Resize image to target size. if keep_ratio is True, 
+        resize the image's long side to the maximum of target_size
+        if keep_ratio is False, resize the image to target size(h, w)
+        Args:
+            target_size (int|list): image target size
+            keep_ratio (bool): whether keep_ratio or not, default true
+            interp (int): the interpolation method
+        """
+        super(ResizeOp, self).__init__()
+        self.keep_ratio = keep_ratio
+        self.interp = interp
+        if not isinstance(target_size, (int, list, tuple)):
+            raise TypeError(
+                "Type of target_size is invalid. Must be Integer or List or Tuple, now is {}".
+                format(type(target_size)))
+        if isinstance(target_size, int):
+            target_size = [target_size, target_size]
+        self.target_size = target_size
+
+    def apply(self, sample, context=None):
+        """ Resize the image numpy.
+        """
+        im = sample['image']
+        if not isinstance(im, np.ndarray):
+            raise TypeError("{}: image type is not numpy.".format(self))
+        if len(im.shape) != 3:
+            raise ImageError('{}: image is not 3-dimensional.'.format(self))
+
+        # apply image
+        im_shape = im.shape
+        if self.keep_ratio:
+
+            im_size_min = np.min(im_shape[0:2])
+            im_size_max = np.max(im_shape[0:2])
+
+            target_size_min = np.min(self.target_size)
+            target_size_max = np.max(self.target_size)
+
+            im_scale = min(target_size_min / im_size_min,
+                           target_size_max / im_size_max)
+
+            resize_h = int(im_scale * im_shape[0])
+            resize_w = int(im_scale * im_shape[1])
+
+            im_scale_x = im_scale
+            im_scale_y = im_scale
+        else:
+            resize_h, resize_w = self.target_size
+            im_scale_y = resize_h / im_shape[1]
+            im_scale_x = resize_w / im_shape[0]
+
+        im = cv2.resize(
+            im,
+            None,
+            None,
+            fx=im_scale_x,
+            fy=im_scale_y,
+            interpolation=self.interp)
+        sample['image'] = im
+        sample['im_shape'] = [resize_h, resize_w]
+        scale_factor = sample['scale_factor']
+        sample['scale_factor'] = [
+            scale_factor[0] * im_scale_y, scale_factor[1] * im_scale_x
+        ]
+
+        # apply bbox
+        gt_bbox = sample['gt_bbox']
+        gt_bbox[:, 0::2] = np.clip(gt_bbox[:, 0::2], 0, resize_w)
+        gt_bbox[:, 1::2] = np.clip(gt_bbox[:, 1::2], 0, resize_h)
+        sample['gt_bbox'] = gt_bbox
```

Comment:
what about gt_mask

Reply:
sample['semantic'] and `sample['gt_segm']` also need here.

Reply:
done, thanks